kclvm/parser/src/lexer/mod.rs

```
@@ -296,47 +296,100 @@ impl<'a> Lexer<'a> {
             kclvm_lexer::TokenKind::Eq => token::Assign,
             // Delim tokens
             kclvm_lexer::TokenKind::OpenParen => {
-                self.indent_cxt.nesting += 1;
+                self.indent_cxt.delims.push(token::OpenDelim(token::Paren));
                 token::OpenDelim(token::Paren)
             }
-            kclvm_lexer::TokenKind::CloseParen => {
-                if self.indent_cxt.nesting == 0 {
-                    self.sess.struct_span_error(
+            kclvm_lexer::TokenKind::CloseParen => match self.indent_cxt.delims.pop() {
+                Some(delim) => match delim {
+                    token::OpenDelim(token::Paren) => token::CloseDelim(token::Paren),
+                    token::OpenDelim(token::Brace) => {
+                        self.sess.struct_span_error_recovery(
+                            "error nesting on close paren",
+                            self.span(start, self.pos),
+                        );
+                        token::CloseDelim(token::Brace)
+                    }
+                    token::OpenDelim(token::Bracket) => {
+                        self.sess.struct_span_error_recovery(
+                            "error nesting on close paren",
+                            self.span(start, self.pos),
+                        );
+                        token::CloseDelim(token::Bracket)
+                    }
+                    _ => token::CloseDelim(token::Paren),
```

Pls using `bug!()` to raise an internal error instead of ignoring this error.